core/src/clients/snyk/snyk_client.rs

```diff
@@ -0,0 +1,223 @@
+        }
+    }
+
+    // Returns a tuple that contains a list of OrgProject objects and a list of any errors encountered
+    async fn get_projects_from_org_list(&self, snyk_token: String, orgs: Vec<Org>) -> (Vec<OrgProjects>, Vec<SnykProviderError>) {
+
+        let mut all_projects: Vec<OrgProjects> = Vec::new();
+        let mut errors: Vec<SnykProviderError> = Vec::new();
+        for org in orgs.iter() {
+            let org_id = org.id.clone().unwrap_or_else(|| "Missing".to_string());
+            let org_name = org.name.clone().unwrap_or_else(|| "Missing".to_string());
+            let results = self.get_projects_from_org(snyk_token.clone(), org_id.clone(), org_name.clone()).await;
+            
+            match results {
+                Ok(option_proj_json) => {
+                    match option_proj_json {
+                        Some(org_proj) => all_projects.push(org_proj),
+                        None => {
+                            let err_msg = SnykProviderError::SnykNoProjectsFoundError(format!("----------------------\nNo projects found for \n--org_id: {}, \n--org_name: {}", org_id, org_name));
+                            errors.push(err_msg);
+                        }
+                    }
+                },
+                Err(err) => errors.push(err),
+            }
+        }
+        return (all_projects, errors);
+    }
+
+
+    // Returns an sbom from Snyk based 
+    async fn get_sbom_from_snyk(&self, snyk_token: String, org_id: String, proj_id: String) -> Result<Option<HashMap<String, Value>>, SnykProviderError> {
+        let sbom_url = format!("{}/orgs/{}/projects/{}/sbom?version={}&format={}", API_V3_URL, org_id, proj_id, API_V3_VERSION, SBOM_FORMAT);
+        let display_msg = format!("----------------------\nFetching sbom from Snyk \nproject: {}, \nOrg: {}, \nUrl: {}", proj_id, org_id, sbom_url);
+        println!("{}", display_msg);
+        let response: Result<Option<HashMap<String, Value>>, HyperError> = get(
+            &sbom_url,
+            ContentType::Json,
+            snyk_token.as_str(),
+            None::<String>,
+        ).await;
+
+        match response {
+            Ok(sbom) => return Ok(sbom),
+            Err(err) => Err(SnykProviderError::SnykSBOMError(format!("Error with: {}, \nadditional errors: {}", display_msg, err)))
+        }
+    }
+
+
+}
+
+#[tokio::test]
+pub async fn test_get_orgs() {
+    let snyk_api = SnykRestClientImpl{};
+    let snyk_token = get_snyk_access_token().await;
+
+    let orgs = snyk_api.get_orgs(snyk_token).await;
+
+    // Validate we have a successfull response
+    assert!(orgs.is_ok(), "Response should be OK");
+
+    // Validate the result has something in it
+    assert!(!orgs.unwrap().is_empty(), "The resulting Vec<Org> should not be empty");
+}
+
+#[tokio::test]
+async fn test_get_projects_from_org() {
+    let snyk_api = SnykRestClientImpl{};
+    let org_id = format!("f288c129-6c28-4b65-aec2-bd753e095b13");
```

Comment:
So while I don't think this counts as a secret, I do think that Sinan's secret scanning work would trigger a warning if it gets implemented before this got merged. Is there a way to dynamically look up the test org?